### PR TITLE
Do not inject web debug toolbar on attachments

### PR DESCRIPTION
**Pull request by peterrehm**
| Q | A |
| --- | --- |
| Branch? | 2.7 |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | #18965 |
| License | MIT |
| Doc PR | - |

### CODE REVIEW

**Comment by javiereguiluz** [comment]
:+1:

Status: reviewed

**Comment by xabbuh** [comment]
👍

File: src/Symfony/Bundle/WebProfilerBundle/EventListener/WebDebugToolbarListener.php [hunk 1]
```diff
@@ -112,6 +112,11 @@ public function onKernelResponse(FilterResponseEvent $event)
      */
     protected function injectToolbar(Response $response, Request $request)
     {
+        // The toolbar shall not be injected if the header enforces a download of the content
+        if (false !== strpos($response->headers->get('Content-Disposition'), 'attachment')) {
```
Comment (fabpot) [inline src/Symfony/Bundle/WebProfilerBundle/EventListener/WebDebugToolbarListener.php]: This should be moved in the condition starting at line 96 instead

Comment (peterrehm) [inline src/Symfony/Bundle/WebProfilerBundle/EventListener/WebDebugToolbarListener.php]: Updated.

File: src/Symfony/Bundle/WebProfilerBundle/EventListener/WebDebugToolbarListener.php [hunk 2]
```diff
@@ -98,6 +98,7 @@ public function onKernelResponse(FilterResponseEvent $event)
             || $response->isRedirection()
             || ($response->headers->has('Content-Type') && false === strpos($response->headers->get('Content-Type'), 'html'))
             || 'html' !== $request->getRequestFormat()
+            || false !== strpos($response->headers->get('Content-Disposition'), 'attachment')
```
Comment (Tobion) [inline src/Symfony/Bundle/WebProfilerBundle/EventListener/WebDebugToolbarListener.php]: The content-disposition types are case-insensitive according to https://tools.ietf.org/html/rfc2183
Also this current check will also trigger for `Content-Disposition: <something-else>; filename=attachment` which is probably not intended.

Comment (peterrehm) [inline src/Symfony/Bundle/WebProfilerBundle/EventListener/WebDebugToolbarListener.php]: Updated accordingly. The question is now the check, I am now checking for `attachment;` instead.

File: src/Symfony/Bundle/WebProfilerBundle/EventListener/WebDebugToolbarListener.php [hunk 3]
```diff
@@ -98,6 +98,7 @@ public function onKernelResponse(FilterResponseEvent $event)
             || $response->isRedirection()
             || ($response->headers->has('Content-Type') && false === strpos($response->headers->get('Content-Type'), 'html'))
             || 'html' !== $request->getRequestFormat()
+            || false !== strpos(strtolower($response->headers->get('Content-Disposition')), 'attachment;')
```
Comment (nicolas-grekas) [inline src/Symfony/Bundle/WebProfilerBundle/EventListener/WebDebugToolbarListener.php]: str**i**pos instead of strpos+strtolower?

Comment (peterrehm) [inline src/Symfony/Bundle/WebProfilerBundle/EventListener/WebDebugToolbarListener.php]: Updated

**Comment by fabpot** [comment]
Thank you @peterrehm.

**Pull request closed without merging**